### PR TITLE
allow explicitly disabling request and session timeouts with 0

### PR DESCRIPTION
**Pull request by gunters63**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
why is it still a draft?

**Comment by gunters63** [comment]
I made a comment in the issue (https://github.com/fastify/fastify-reply-from/issues/366), I maybe should have better put it here.

The PR was still a draft because I was unsure about the missing test about disabling the sessionTimeout. I wrote a test, but it never completed and tap ran into a timeout. I don't really know what is the reason. So i left the test out.

Maybe this isn't a problem, so I'll undraft the PR now.

**Comment by mcollina** [comment]
Ah I see it. So a test is missing for `requestTimeout`? It seems the one for `sessionTimeout` is there?

File: test/http2-timeout-disabled.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      }),
+      new Promise((resolve, reject) => setTimeout(reject, 11000, 'passed'))
+    ])
+  } catch (err) {
+    // if we wait 11000ms without a timeout error, we assume disabling the timeout worked
+    // 10000 ms is the default timeout
+    t.equal(err, 'passed')
+
+    return
+  }
```
Comment (Eomm) [inline test/http2-timeout-disabled.test.js:45]:
```suggestion
    const result = await Promise.race([
      got.get(`http://localhost:${instance.server.address().port}/`, {
        retry: 0
      }),
      new Promise((resolve) => setTimeout(resolve, 11000, 'passed'))
    ])

    // if we wait 11000ms without a timeout error, we assume disabling the timeout worked
    // 10000 ms is the default timeout
    t.equal(result, 'passed')
```

**Comment by gunters63** [comment]
> Ah I see it. So a test is missing for `requestTimeout`? It seems the one for `sessionTimeout` is there?

The other way around :)

File: test/http2-timeout-disabled.test.js (hunk 1 above)
Comment (gunters63) [reply test/http2-timeout-disabled.test.js:45]: Ah, yes, good point. race will work better here.

**Comment by mcollina** [comment]
Can you add the test about sessionTimeout?

**Comment by gunters63** [comment]
Ok, I think I solved the problem with the hanging test. I added a proper disabling session timeout test now.

I explicitly clean up the request and the two Fastify instances directly after the timeout. Then the test finishes cleanly. If I leave out any of this:

```js
  // clean up right after the timeout, otherwise test will hang
  request.cancel()
  target.close()
  instance.close()
```

the test hangs. Cleaning at teardown time (t.teardown) does not help here, maybe a chicken/egg problem.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**